File: ExpressControls/FontItems.vb
Imports System.Collections.ObjectModel
Imports WinDrawing = System.Drawing

Public Class FontItems
    Inherits ObservableCollection(Of String)

    Public Sub New()
        Dim objFontCollection As WinDrawing.Text.FontCollection = New WinDrawing.Text.InstalledFontCollection

        For Each objFontFamily As WinDrawing.FontFamily In objFontCollection.Families
            Dim fontname As String = objFontFamily.Name

            If Not fontname = "" Then
                Add(fontname)
            End If
        Next
    End Sub

End Class

File: ExpressControls/FontItems.cs
using System.Collections.ObjectModel;
using System.Linq;
using WinDrawing = System.Drawing;

namespace ExpressControls
{
    public class FontItems : ObservableCollection<string>
    {
        public FontItems() { }

        public FontItems(bool load = false)
        {
            if (load)
            {
                WinDrawing.Text.FontCollection objFontCollection =
                    new WinDrawing.Text.InstalledFontCollection();

                foreach (
                    string objFontFamily in objFontCollection
                        .Families.Select((el) => el.Name)
                        .Where((el) => !string.IsNullOrEmpty(el))
                )
                {
                    Add(objFontFamily);
                }
            }
        }
    }
}
